text/python/0000-dependencies-update.md

```diff
@@ -0,0 +1,97 @@
+# Decide which Python dependencies will be Paketo-hosted
+
+## Proposal
+
+Currently four Python buildpacks have Paketo-hosted dependencies.
+Only two Paketo-hosted dependencies should be kept, and two should be removed.
+
+| Buildpack                                                                        | Action |
+|----------------------------------------------------------------------------------|--------|
+| [cpython](https://github.com/paketo-buildpacks/cpython/blob/main/buildpack.toml) | Keep   |
+| [poetry](https://github.com/paketo-buildpacks/poetry/blob/main/buildpack.toml)   | Remove |
+| [pip](https://github.com/paketo-buildpacks/pip/blob/main/buildpack.toml)         | Keep   |
+| [pipenv](https://github.com/paketo-buildpacks/pipenv/blob/main/buildpack.toml)   | Remove |
+
+### Offline not supported
+
+Note that the Python language family does not support a fully-offline case, since several of its components require internet connectivity to work.
+These include at least the following: 
+
+* pipenv: [#limitations](https://github.com/paketo-buildpacks/pipenv#limitations)
+* poetry: [#known-issues-and-limitations](https://github.com/paketo-buildpacks/poetry#known-issues-and-limitations)
+
+## Rationale
+
+### CPython
+
+Keep this as a Paketo-hosted dependency.
+
+Currently this uses `dep-server`, `binary-builder`, and `buildpacks-ci` to build, but language family maintainers
+will transition this to the new Github Action workflow once that has been approved.
+
+### Poetry
+
+Remove the Paketo-hosted dependency.
+
+This dependency seems to be the result of a `pip download` command as seen in [buildpacks-ci](https://github.com/cloudfoundry/buildpacks-ci/blob/6e8873b5a1535c2faf56f810fde7063864db7585/tasks/build-binary-new/builder.rb#L98-L122).
+This could easily be replicated in the buildpack's `build` stage.
+
+As of 2022-06-28, this buildpack is actually not using the dependency from its `buildpack.toml`.
+Instead, the buildpack is running `pip download poetry==$VERSION`, see its [source code](https://github.com/paketo-buildpacks/poetry/blob/ca9e73b34d7018ce00f7d75c76382a08df554a41/poetry_install_process.go#L35).
+
+This RFC does not require Python maintainers to incorporate all logic from `buildpacks-ci` into the `poetry` buildpack.
+
+Also, see [Offline not supported](#offline-not-supported).
+
+#### Poetry layer caching and reproducibility
+
+Remove support for `poetry` layer caching and image reproducibility.
```

Comment:
This seems like a pretty hallmark feature of buildpacks. Is there no way to consume Poetry another way? Or to prevent the `pip download` to be skipped on rebuilds?

Reply:
I've updated the language to reflect that we want to know (via the `buildpack.yaml[metadata.default-versions]`) what version of the dependency will be installed, and that the dependency layer will be preserved for reuse.